From 6005102330746bbf84c1bede51c8d04c93241897 Mon Sep 17 00:00:00 2001
From: Mykola Ostrovskyy <mykola@ti.com>
Date: Tue, 8 Apr 2014 14:36:57 -0500
Subject: [PATCH 09/11] HWComposer: Use native FB format for HWC 1.1

Until HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED is supported by
EGL, we have to hard-code FB pixel format.

Change-Id: I0f654c894bbfa82851bb03b85ad59e7c7305f309
Signed-off-by: Mykola Ostrovskyy <mykola@ti.com>
---
 services/surfaceflinger/DisplayHardware/HWComposer.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/services/surfaceflinger/DisplayHardware/HWComposer.cpp b/services/surfaceflinger/DisplayHardware/HWComposer.cpp
index d67dab3..a3abad2 100644
--- a/services/surfaceflinger/DisplayHardware/HWComposer.cpp
+++ b/services/surfaceflinger/DisplayHardware/HWComposer.cpp
@@ -431,7 +431,12 @@ status_t HWComposer::queryDisplayProperties(int disp) {
     }
 
     // FIXME: what should we set the format to?
+#ifdef OMAP_ENHANCEMENT
+    // Use pixel format native to DSS HW
+    mDisplayData[disp].format = HAL_PIXEL_FORMAT_BGRA_8888;
+#else
     mDisplayData[disp].format = HAL_PIXEL_FORMAT_RGBA_8888;
+#endif
     mDisplayData[disp].connected = true;
     return NO_ERROR;
 }
@@ -855,7 +860,12 @@ int HWComposer::getVisualID() const {
         // FIXME: temporary hack until HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED
         // is supported by the implementation. we can only be in this case
         // if we have HWC 1.1
+#ifdef OMAP_ENHANCEMENT
+        // Use pixel format native to DSS HW
+        return HAL_PIXEL_FORMAT_BGRA_8888;
+#else
         return HAL_PIXEL_FORMAT_RGBA_8888;
+#endif
         //return HAL_PIXEL_FORMAT_IMPLEMENTATION_DEFINED;
     } else {
         return mFbDev->format;
-- 
2.7.4

